ppx/reason_react_ppx.ml

```diff
@@ -641,29 +641,6 @@ let jsxMapper =
               API instead.")
     | Pexp_fun (arg, default, pattern, expression)
       when isOptional arg || isLabelled arg ->
-        let () =
-          match (isOptional arg, pattern, default) with
-          | true, { ppat_desc = Ppat_constraint (_, { ptyp_desc }) }, None -> (
-              match ptyp_desc with
-              | Ptyp_constr ({ txt = Lident "option" }, [ _ ]) -> ()
-              | _ ->
-                  let currentType =
-                    match ptyp_desc with
-                    | Ptyp_constr ({ txt }, []) ->
-                        String.concat "." (Longident.flatten_exn txt)
-                    | Ptyp_constr ({ txt }, _innerTypeArgs) ->
-                        String.concat "." (Longident.flatten_exn txt) ^ "(...)"
-                    | _ -> "..."
-                  in
-                  Ocaml_common.Location.prerr_warning pattern.ppat_loc
-                    (Preprocessor
-                       (Printf.sprintf
-                          "react-jsx-ppx: optional argument annotations must \
```

Comment:
Should we first add some tests that exercise this code path, then apply this change to see how they are affected? it's impossible for me to know what this change is doing by just looking at the code. My understanding is that it was making sure that optional labelled args in props are properly annotated, right?

Reply:
Adding tests is a great idea. Will do that to help us visualize the coverage better 👍